### DISCUSSION TITLE
Authentication Not Working Even with browser

### DISCUSSION THREAD

**Discussion by brainstorm-tech** [homebridge-simplisafe3/homebridge-simplisafe3#359] [General]
Unfortunately, for the past 48 hours I cannot authenticate, even with the Firefox or Chrome browsers.  I am using plugin v1.10.8 and Homebridge UI 4.50.1.  MacOS 12.5.

I am seeing the Error 400 for authentication, and trying to use the new (older) browser authentication process.

Clicking anything in the plugin at all gives me the "Terminating child process..." and the process is gone.  It doesn't seem to give me the chance to log in manually, but when I do, and paste the URL, it says "Error retrieving token".  But by that point the child process is gone.

Any recommendations?  I can't seem to find any way to keep the process alive long enough to authenticate, although I have no problems logging into the simplisafe page directly.

Thanks,
-Ed

**Comment by thepennster** [comment]
I have the exact same problem, and I've tried authenticating across various browsers, but it seems to hang on the 2FA page. Once I input the code, the page does not refresh and the webpage just hangs there.

**Reply by dustinprhodes** [reply]
Exact same issue

**Reply by shamoon** [reply]
Are you looking in the error console for the final URL? No browser will actually forward you there, just show an error and you need to copy the URL

**Comment by shamoon** [comment]
Look in the error console e.g. final screen should look like
<img width="1435" alt="Screen Shot 2022-08-13 at 7 20 55 AM" src="https://user-images.githubusercontent.com/4887959/184498446-96e13ef1-4098-44eb-8dcd-ed201471ce21.png">

It's that URL `com.simplisafe.mobile://...` you need to copy and paste

**Reply by thepennster** [reply]
> Look in the error console e.g. final screen should look like <img alt="Screen Shot 2022-08-13 at 7 20 55 AM" width="1435" src="https://user-images.githubusercontent.com/4887959/184498446-96e13ef1-4098-44eb-8dcd-ed201471ce21.png">
> 
> It's that URL `com.simplisafe.mobile://...` you need to copy and paste

I just tried again and for me at least, the page doesn't refresh after inputing the 2FA code, it just hangs there and I am not being redirected. I wonder if the issue is with SS.

**Reply by thepennster** [reply]
Here is the error I get in the browser: 

Failed to launch 'com.simplisafe.mobile://auth.simplisafe.com/ios/com.simplisafe.mobile/callback?code=SOMERANDOMCODEHERE because the scheme does not have a registered handler.

**Reply by shamoon** [reply]
Yes, please look at the screenshot / re-read the instructions. Copy that URL and thats what you paste into the plugin config.

**Reply by shamoon** [reply]
Tested with SMS too, same thing, URL is in the console, works fine.

<img width="1435" alt="Screen Shot 2022-08-13 at 7 44 30 AM" src="https://user-images.githubusercontent.com/4887959/184499255-4ea18e26-7853-4e74-b13a-25dd939dc57f.png">

**Reply by thepennster** [reply]
Thank you, that worked for me. I read the instructions, and misunderstood step iii in the instructions where I expected to be redirected to another page or error page to retrieve the URL after inputting the 2FA code. Thank you again!

**Reply by jayali786** [reply]
i copy this "'com.simplisafe.mobile://auth.simplisafe.com/ios/com.simplisafe.mobile/callback?code=SOMERANDOMCODEHERE" browser but all i get is error message and no other url link

help me how you solve this issue

**Reply by jayali786** [reply]
This site can’t be reachedCheck if there is a typo in com.simplisafe.mobile.
If spelling is correct, try running Windows Network Diagnostics.
DNS_PROBE_FINISHED_NXDOMAIN 

this what i get

**Reply by shamoon** [reply]
Please red the output in the original discussion you opened. You have a completely different issue

**Comment by brainstorm-tech** [comment]
The URL-copy mechanism never did work for me.  But plugin 1.10.9 worked perfectly - thank you!!!

**Comment by four43** [comment]
For what it's worth - I just setup my plugin with 2FA text codes and it all worked great from within the Homebridge interface.
